### PR TITLE
US003 Set Item Done

### PR DESCRIPTION
**Pull request by Zeroya**
Add logic for checkbox

### CODE REVIEW

File: src/components/TodoItem/TodoItem.tsx [hunk 1]
```diff
@@ -1,17 +1,34 @@
 import React from "react";
 import { ITodo } from "../../models/ITodo";
+import { completeTodo } from "../../store/reducers/UserSlice";
+import { useAppDispatch } from "../../hooks/hooks";
 import s from "./TodoItem.module.scss";
 
-const TodoItem: React.FC<ITodo> = ({ message, date, dateExpiration }) => {
+const TodoItem: React.FC<ITodo> = ({ id, message, date, dateExpiration, completed }) => {
```
Comment (Kost927) [inline src/components/TodoItem/TodoItem.tsx]: you can write just FC, without React, FC has to be imported from React. import { FC } from 'react"

File: src/components/TodoItem/TodoItem.tsx [hunk 2]
```diff
@@ -1,17 +1,34 @@
...
+  const dispatch = useAppDispatch();
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
```
Comment (Kost927) [inline src/components/TodoItem/TodoItem.tsx:10]: The same with ChangeEvent

File: src/components/TodoItem/TodoItem.tsx [hunk 3]
```diff
@@ -1,17 +1,34 @@
...
+        <p className={`${completed ? s.completed : ""}`}>{message}</p>
+      </div>
+      <div className={`${s.todoItem_expData} ${s.todoItem_textColor} ${completed ? s.completed : ""}`}>
+        <p>Creation date</p>
+        <p>{date}</p>
       </div>
-      <p>{date}</p>
-      <div className={s.todoItem_expData}>
-        <p>Expiration data</p>
+      <div className={`${s.todoItem_expData} ${completed ? s.completed : ""}`}>
```
Comment (Kost927) [inline src/components/TodoItem/TodoItem.tsx]: the same logic " completed ? s.completed : "" " more than 2 times. Create const *** = completed ? s.completed : ""  before return and put ***  variable in JSX.

Comment (Kost927) [reply src/components/TodoItem/TodoItem.tsx]: And the good practice is to not put login into JSX.

File: src/components/TodoItem/TodoItem.tsx (hunk 1 above)
Comment (Zeroya) [reply src/components/TodoItem/TodoItem.tsx]: Yes, I know about this, I will try to add FC to import further. Сhanged this moment in the project

File: src/components/TodoItem/TodoItem.tsx (hunk 2 above)
Comment (Zeroya) [reply src/components/TodoItem/TodoItem.tsx:10]: Сhanged this moment in the project

File: src/components/TodoItem/TodoItem.tsx (hunk 3 above)
Comment (Zeroya) [reply src/components/TodoItem/TodoItem.tsx]: I moved the class logic to a separate variable before return

**Review by Kost927 — APPROVED** [review]

**Pull request merged**